spring-jdbc/documentation.md

```diff
@@ -0,0 +1,188 @@
+# The Crate JDBC Sample Application
+
+**Note:** The application is written in Java 11.
+
+## Installation
+
+Staring from 0.57.0 Crate supports for the [PostgreSQL wire protocol v3][1].
+To get more detail information on the protocol support by Crate,
+please have a look at the [documentation][2].
+
+In the sample application, we use the [Crate JDBC driver 2.1.7][3] which
+uses [PostgreSQL wire protocol v3][1]. To obtain the [Crate JDBC driver][4],
+please follow [instructions][5] for the build tool of your choice.
+
+## Usage
+
+### Connecting to Crate
+
+The application establishes the connection with Crate using a Spring JDBC datasource configured in the `application.properties` file:
+
+```text
+spring.datasource.url=jdbc:crate://localhost:5432/
+spring.datasource.username=crate
+spring.datasource.password=
+spring.datasource.driver-class-name=io.crate.client.jdbc.CrateDriver
+```
+
+Please, take a look at the Crate JDBC driver documentation to see the
+[possible forms][6] of URLs supported by the driver.
+
+### Executing statements
+
+For all statements Spring's JDBC template is used. E.g reading a blog post from CrateDB:
+
+```java
+    public Optional<BlogPost> getPost(@NonNull String id) {
+
+        List<BlogPost> queryResult = this.jdbcTemplate.query(String.format(
+                "SELECT p.id as id, p.text as text, "
+                        + "p.\"user\"['name'] as username, p.\"user\"['location'] as userlocation, "
+                        + "p.created as created, c.name as country, c.geometry as area, "
+                        + "p.image_ref as imageRef, p.like_count as likes " + "FROM %s AS p, %s AS c "
+                        + "WHERE within(p.\"user\"['location'], c.geometry) " + "AND p.id = ?",
+                POST_TABLE, COUNTRIES_TABLE), (rs, rowNum) -> stdRowMapper.apply(rs), id);
```

Comment:
Is there a specific reason why the table names are not part of the statement itself ?

Reply:
Just did it the same way as the original JDBC sample

Reply:
And btw. I think it's a good idea to have the table names not part of the statement. If you want to change the table names, you'll have to change tens of statements correctly!